pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -1198,18 +1198,18 @@ func (c *PodAffinityChecker) podMatchesPodAffinityTerms(pod, targetPod *v1.Pod,
 		return false, false, false, nil
 	}
 	// Namespace and selector of the terms have matched. Now we check topology of the terms.
-	sameToploykeyNum := 0
+	matchedTermCount := 0
 	for _, term := range matchedTerms {
 		if len(term.TopologyKey) == 0 {
 			return false, false, false, fmt.Errorf("empty topologyKey is not allowed except for PreferredDuringScheduling pod anti-affinity")
 		}
 		if priorityutil.NodesHaveSameTopologyKey(nodeInfo.Node(), targetPodNode, term.TopologyKey) {
-			sameToploykeyNum = sameToploykeyNum + 1
+			matchedTermCount++
 		}
 	}
-	if sameToploykeyNum == len(matchedTerms) {
+	if matchedTermCount == len(matchedTerms) && matchedTermCount == podAffinityNum {
```

Comment:
IIUC, podAffinityNum equals to len(matchedTerms), so "matchedTermCount == podAffinityNum" is unnecessary, right?

Reply:
This is why the func getMatchingAffinityTerms has a return value podAffinityNum, for example, pod has two affinity terms which is affinity "foo":"bar" and "disk":"ssd", and the targetpod has two label "foo":bar，“disk”:"slow",so the matchedTerm only has one "foo":"bar", for the affinity "match all" semantic, we should have a "matchedTermCount == len(podAffinityNum) " to verify it

Reply:
Ok,i will commit later~